csharp/test/Apache.Arrow.Tests/CDataInterfaceSchemaTests.cs

```diff
@@ -0,0 +1,302 @@
+                yield return pa.field("i8", pa.int8(), true);
+                yield return pa.field("u8", pa.uint8(), true);
+                yield return pa.field("i16", pa.int16(), true);
+                yield return pa.field("u16", pa.uint16(), true);
+                yield return pa.field("i32", pa.int32(), true);
+                yield return pa.field("u32", pa.uint32(), true);
+                yield return pa.field("i64", pa.int64(), true);
+                yield return pa.field("u64", pa.uint64(), true);
+
+                yield return pa.field("f16", pa.float16(), true);
+                yield return pa.field("f32", pa.float32(), true);
+                yield return pa.field("f64", pa.float64(), true);
+
+                yield return pa.field("binary", pa.binary(), false);
+                yield return pa.field("string", pa.utf8(), false);
+
+                yield return pa.field("date32", pa.date32(), false);
+                yield return pa.field("date64", pa.date64(), false);
+                yield return pa.field("time32_s", pa.time32("s"), false);
+                yield return pa.field("time32_ms", pa.time32("ms"), false);
+                yield return pa.field("time64_us", pa.time64("us"), false);
+                yield return pa.field("time64_ns", pa.time64("ns"), false);
+
+                yield return pa.field("list_string", pa.list_(pa.utf8()), false);
+                yield return pa.field("list_list_i32", pa.list_(pa.list_(pa.int32())), false);
+
+                yield return pa.field("dict_string", pa.dictionary(pa.int32(), pa.utf8(), false), false);
+                yield return pa.field("dict_string_ordered", pa.dictionary(pa.int32(), pa.utf8(), true), false);
+                yield return pa.field("list_dict_string", pa.list_(pa.dictionary(pa.int32(), pa.utf8(), false)), false);
+            }
+        }
+
+        public static dynamic GetPythonSchema()
+        {
+            using (Py.GIL())
+            {
+                dynamic pa = Py.Import("pyarrow");
+                return pa.schema(GetPythonFields().ToList());
+            }
+        }
+
+        // Schemas created in Python, used in CSharp
+        [Fact]
+        public void ImportType()
+        {
+            PythonEngine.Initialize();
+            Schema schema = GetTestSchema();
+            IEnumerable<dynamic> py_fields = GetPythonFields();
+
+            foreach ((Field field, dynamic py_field) in schema.Fields.Values.AsEnumerable()
+                .Zip(py_fields))
+            {
+                var c_schema = new CArrowSchema();
+                IntPtr imported_ptr = c_schema.AllocateAsPtr();
+
+                using (Py.GIL())
+                {
+                    dynamic py_datatype = py_field.type;
+                    py_datatype._export_to_c(imported_ptr.ToInt64());
+                }
+
+                var dataTypeComparer = new ArrayTypeComparer(field.DataType);
+                var imported_type = new ImportedArrowSchema(imported_ptr);
```

Comment:
How about using the same naming convention? (`camelCase` v.s. `snake_case`)

Reply:
That's a good point. I need to fix the code style.